src/sentry/release_health/duplex.py

```diff
@@ -0,0 +1,763 @@
+import collections.abc
+import inspect
+from copy import deepcopy
+from datetime import datetime, timedelta
+from enum import Enum
+from typing import Any, Callable, List, Mapping, Optional, Sequence, Set, TypeVar, Union
+
+from dateutil import parser
+from typing_extensions import Literal
+
+from sentry.release_health.base import (
+    CrashFreeBreakdown,
+    CurrentAndPreviousCrashFreeRates,
+    EnvironmentName,
+    OrganizationId,
+    OverviewStat,
+    ProjectId,
+    ProjectOrRelease,
+    ProjectRelease,
+    ProjectReleaseSessionStats,
+    ProjectReleaseUserStats,
+    ProjectWithCount,
+    ReleaseHealthBackend,
+    ReleaseHealthOverview,
+    ReleaseName,
+    ReleasesAdoption,
+    ReleaseSessionsTimeBounds,
+    SessionsQueryResult,
+    StatsPeriod,
+)
+from sentry.release_health.metrics import MetricsReleaseHealthBackend
+from sentry.release_health.sessions import SessionsReleaseHealthBackend
+from sentry.snuba.sessions import get_rollup_starts_and_buckets
+from sentry.snuba.sessions_v2 import QueryDefinition
+
+DateLike = TypeVar("DateLike", datetime, str)
+
+
+class ComparatorType(Enum):
+    Counter = "counter"
+    Ratio = "ratio"
+    Quantile = "quantile"
+    Entity = "entity"
```

Comment:
Would it make sense to rename `Entity` to `Exact`?

Reply:
Yes, makes sense, let me think a bit on int